react-native-template-reason/src/App.re

```diff
@@ -4,55 +4,50 @@
  *
  * Converted from https://github.com/facebook/react-native/blob/d752446b23e0a12171ca0c01d20f5be625f40455/template/App.js
  */
-open BsReactNative;
+open ReactNative;
 
-let instructions =
-  switch (Platform.os()) {
-  | IOS(_) => "Press Cmd+R to reload,\n" ++ "Cmd+D or shake for dev menu"
-  | Android =>
-    "Double tap R on your keyboard to reload,\n"
-    ++ "Shake or press menu button for dev menu"
-  | exception (Platform.UnknownPlatform(p)) when p == "web" => "Press Cmd+R to reload."
-  | exception (Platform.UnknownPlatform(_p)) => ""
-  };
+module Styles = {
+  open Style;
 
-let styles =
-  Style.(
-    StyleSheet.create({
-      "container":
-        style([
-          flex(1.),
-          justifyContent(Center),
-          alignItems(Center),
-          backgroundColor(String("#F5FCFF")),
-        ]),
-      "welcome":
-        style([
-          fontSize(Float(20.)),
-          textAlign(Center),
-          margin(Pt(10.)),
-        ]),
-      "instructions":
-        style([
-          textAlign(Center),
-          color(String("#333333")),
-          marginBottom(Pt(5.)),
-        ]),
-    })
+  let container = style(
+    ~flex=1.,
+    ~justifyContent=`center,
+    ~alignItems=`center,
+    ~backgroundColor="#F5FCFF"
   );
 
-let app = () =>
-  <View
-    style={
-      styles##container;
-    }>
-    <Text style={styles##welcome}>
-      "Welcome to (Bs) React Native!"->ReasonReact.string
+  let welcome = style(
+    ~fontSize=20.,
+    ~textAlign=`center,
+    ~margin=pt(10.)
+  );
+
+  let instructions = style(
+    ~textAlign=`center,
+    ~color="#333333",
+    ~marginBottom=pt(5.)
+  );
+};
```

Comment:
Can you revert this to an actual StyleSheet? It's better for performance so we should definitely not promote this kind of modules for styling. Imo it's a bad practise I see too often in bs-rn world.

Reply:
Sure I can put Stylesheet instead. When we started to work with bs-react-native we hesitated between the 2 versions. We didn't use Stylesheet because we didn't understood how it was beneficial. Can you explain to me this perfomance issue you're talking about?

Reply:
When you use a module like that, you just send raw styles. StyleSheet on the other hand prepare styles & create internal identifiers (that's why StyleSheet elements are `int` in JS). Optimization can be made from there in native part (I am not 100% sure that's the case, I am guessing here).

The fact that StyleSheet.flatten have an explicit cost (see https://facebook.github.io/react-native/docs/stylesheet#flatten) make me think that an optimisation is made. Also if you have some redundant styles, a unique id will be used (and this is very likely to be the case for reusable components).

For react-native-web I am sure there is a direct benefit when using StyleSheet, so again, pretty sure (but I never checked - to be honest) that's the same for ios & android implementation.

Reply:
I didn't know any of this. Thank you for the explanation.